pkgs/applications/networking/cluster/k3s/default.nix

```diff
@@ -223,7 +221,6 @@ buildGoModule rec {
   version = k3sVersion;
 
   src = k3sRepo;
-  proxyVendor = true;
```

Comment:
When I originally added this I left a note: https://github.com/NixOS/nixpkgs/pull/158089#discussion_r799256819

> Still seems to be required. I couldn't figure out an alternative.

I might have been mistaken then, or something else might have changed :man_shrugging:.

Either way, we can get rid of that, which has the benefit of avoiding an issue with `go mod download` not working on the current k3s release https://github.com/k3s-io/k3s/pull/5548

It simplifies the update script too, woo!